receiver/signalfxreceiver/receiver.go

```diff
@@ -221,40 +238,49 @@ func (r *sfxReceiver) handleDatapointReq(resp http.ResponseWriter, req *http.Req
 		return
 	}
 
-	body, ok := r.readBody(ctx, resp, req)
+	body, otlpFormat, ok := r.readBody(ctx, resp, req, false)
 	if !ok {
 		return
 	}
 
-	msg := &sfxpb.DataPointUploadMessage{}
-	if err := msg.Unmarshal(body); err != nil {
-		r.failRequest(ctx, resp, http.StatusBadRequest, errUnmarshalBodyRespBody, err)
-		return
+	r.settings.Logger.Debug("Handling metrics data")
+
+	var md pmetric.Metrics
+	var err error
```

Comment:
The var seems to be unused. `err = r.metricsConsumer.ConsumeMetrics(ctx, md)` below can be updated instead